### ISSUE TITLE
Compose the `syslog` source with the `socket` and `remap` transform

### ISSUE THREAD

**Issue by binarylogic** [vectordotdev/vector#4502]
Once the work described in https://github.com/timberio/vector/pull/4427 is complete, we should rebuild our `syslog` source in a way that takes advantage of these new changes. The result should be the same as:

```toml
[sources.socket]
type = "socket"
# ...

[transforms.remap]
type = "remap"
inputs = ["socket"]
source = """
parse_syslog(.message)
"""
```

## Requirements

- [ ] Log a warning when parsing fails, like we do now.
- [ ] Ensure that all syslog keys are in the root of the event.

**Comment by binarylogic** [comment]
Just noting, this will be our first endeavor into using the new macros/facade work that @Hoverbear unblocked. We should have a quick discussion around how this will work and if we want to do it.

**Comment by Hoverbear** [comment]
I'm so excited!!!!

**Comment by lukesteensen** [comment]
We should definitely do this. I'm not sure if it literally needs to go through remap or not (we could just have the same underlying implementation), but it would be great to clean up these implementations a whole bunch and exercise our new compositional powers.

**Comment by Hoverbear** [comment]
While we don't have a `Source` trait, we do have a `SourceConfig` trait: https://github.com/timberio/vector/blob/a894cf7cc5370f0d7176dc11a7b93838a045a2c9/src/config/mod.rs#L187-L207

Since this can return any source from `build`, we could have say a `SysLogConfig` whose `build` returns a `socket` source and then uses a `map` call over a remap function. 🤔 

Is the intent to have these all use remap, or will some use a different transform? If we limit ourselves to `remap` or even `FunctionTransform`s we could do this, but if we need `TaskTransform`s there is some added complexity I think.

**Comment by prognant** [comment]
As per https://github.com/timberio/vector/pull/7046#issuecomment-824784967, waiting for the codec feature seems preferable, but outside this decoding layer that is mandatory to properly frame syslog message before further processing, there seems to be no particular pain point.

**Comment by dsmith3197** [comment]
This is done / no longer relevant thanks to Vector's codecs architecture.
